lib/generators/rolemodel/good_job/good_job_generator.rb

```diff
@@ -0,0 +1,74 @@
+require_relative '../../bundler_helpers'
+
+module Rolemodel
+  class GoodJobGenerator < Rails::Generators::Base
+    include Rolemodel::BundlerHelpers
+    source_root File.expand_path('templates', __dir__)
+
+    def install_good_job
+      say 'Install GoodJob'
+
+      bundle_command 'add good_job'
+      run_bundle
+
+      generate 'good_job:install'
+    end
+
+    def configure_good_job
+      say 'Add GoodJob configuration'
+
+      inject_into_file 'config/application.rb', after: "class Application < Rails::Application\n" do
+        optimize_indentation <<~'RUBY', 4
+          # Add GoodJob ActiveJob queue adapter
+          config.active_job.queue_adapter = :good_job
+
+        RUBY
+      end
```

Comment:
Is the preference to modify `application.rb` or include these kinds of settings in a file in `config/initializers.rb`? I've typically done the latter in most cases. I think we've found that can be more reliable for generators, too.

Reply:
This approach [follows the README](https://github.com/bensheldon/good_job?tab=readme-ov-file#set-up).

Reply:
@kylesmile that is a good observation. In our context an initializer probably makes more sense.

Reply:
@kylesmile Yeah, we were trying to follow the documentation so people would know where to look if they read the GoodJob docs. I'm open to moving it but in checking our RoleModel projects, only 2 (landone-takeoff and thundercloud) put it in an initializer. Everything else (141 instances) are either in config/application.rb (11) or config/environments/<something> (130) files because that's where Rails puts it by default.

Reply:
For what its worth, their docs indicate that you can keep it in the initializer if you change it from `config.active_job.queue_adapter = :good_job` to `ActiveJob::Base.queue_adapter = :good_job`. I didn't try that out though to test if it solved my bug since the other way worked.

Reply:
> @jdmcleod [found out](https://rolemodelsoftware.slack.com/archives/C051DL755/p1729690474673799?thread_ts=1729606215.178869&cid=C051DL755) that putting `config.active_job.queue_adapter = :good_job` into an initializer, it caused issues with the jobs running inline instead of in the Heroku worker process when cron was added. We should consider moving that in [thundercloud](https://github.com/rolemodel/thundercloud) @tingraldi @seanpk too.

This isn't biting us right now in thundercloud, but it would be good to relocate the config to avoid potential issues in  the future. I've added an [issue](https://linear.app/rolemodelsoftware/issue/DPQ-343/relocate-good-job-setting) to Linear for this.